### PR TITLE
Add endpoints in Octopoes for bulk operations in the object list page

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
This PR adds two endpoints in Octopoes (that are also being used in Rocky):
- POST `/{client}/objects/delete_many`
- POST `/{client}/scan_profiles/save_many`

These are now used for the object list page where there is a multiselect for OOIs, where we first made an HTTP call for each selected object. In Octopoes these are wrapped in one unit of work for as well, so for N objects this save N - 1 API calls to Octopoes and N - 1 calls from Octopoes to XTDB. The bad performance is currently noticeable even when just the first page of objects is filled and you did a select all on both operations. 

I also added a new integration test layer that tests the OctopoesAPIConnector against an Octopoes api instance (see tests).

### Issue link
Fixes https://github.com/minvws/nl-kat-coordination/issues/228

### Proof
I also added some tags to the Swagger docs:
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by Donnype** [comment]
The tests in the pipeline are way to flaky with the current xtdb-multinode bug.. Perhaps we should disable the tests for now and trust that they work for me locally until the bug has been fixed?

**Comment by Donnype** [comment]
Issue has been fixed!

**Comment by underdarknl** [comment]
> I definitely agree with batch processing such calls. But I don't really see the need for new endpoints. Since the resources are already plurally oriented, we could easily support single/ multiple creation and deletion with the same endpoint. E.g.
> 
> ```
> DELETE /{client}/objects with 
> POST /{client}/scan_profiles
> ```
> 
> The body format (for mostly "create" endpoints) could be `ndjson` to enable passing multiple objects and at the same time keep backward compatibility. Query strings could also work, but many clients and servers implement different maximum sizes which could cause trouble.

I would advice against query string for anything other than single items (and read operations).
We could specify any send data format by sending the Content-Type header with the request, and default to something that is plain/text like we have now, expecting only a single item.

File: octopoes/.ci/docker-compose.yml [hunk 1]
```diff
@@ -38,11 +38,40 @@ services:
     command: pytest -x tests/integration
     depends_on:
       - xtdb_multinode
+      - ci_octopoes
     env_file:
       - ".ci/ci-multinode.env"
 
   xtdb_multinode:
-    image: "ghcr.io/dekkers/xtdb-http-multinode:v1.0.4"
+    image: "ghcr.io/dekkers/xtdb-http-multinode:pr-11"
```
Comment (Darwinkel) [inline octopoes/.ci/docker-compose.yml:46]: Shouldn't this become `ghcr.io/dekkers/xtdb-http-multinode:v1.0.6` now that the PR has been merged?

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

I haven't personally tested the endpoints.

### What works:
* No apparent regressions

**Comment by Donnype** [comment]
@ammar92 @underdarknl To do this the nicest way I think the endpoints regarding a single object should have the primary key as a path parameter instead of a query parameter, but if my memory serves me well this is not possible with the current primary key encoding of the objects (?) and hence the scan profiles. But even in that case it is not advised to either put a lot of primary keys in query strings or a add a body to a DELETE request.

Since there was no POST endpoint for the scan profiles yet we would indeed be able to leverage that one, the risk being that we are obscuring functionality that differs from that what might be assumed for a typical set of REST endpoints (i.e. using POST to save a single object). 

All in all I don't see the benefit of less endpoints if it's at the cost of consistency in the API semantics. I _do_ see the benefit in standardizing bulk operations (see [this](https://backendless.com/docs/rest/data_multiple_objects_bulk_create.html) example) or deciding that every POST endpoint should accept ndjson and support bulk by default. But in the latter case you will into issues with DELETE requests not having request bodies.

**Comment by praseodym** [comment]
+1 for using ndjson as a bulk request content type. Because we don't really follow REST anyway, we might as well switch all DELETE requests to POST too so we can use a request body there too?

**Comment by dekkers** [comment]
What is the benefits of using ndjson instead of just using a list in json? I don't really see them, but see some downsides. We would create inconsistency in the APIs or need to do a lot of work to convert every API to ndjson. Dumping and parsing would be more complicated as we can't use use the standard fastapi / request json support. It would probably also be slower because the jsonlines library is in pure Python while json is implemented as a C extension.

As discussed in the standup yesterday this is also something we easily refactor later because the APIs are only called by other OpenKAT code so I am moving this to ready to merge.

**Comment by praseodym** [comment]
Lack of support in fastapi is indeed a big issue. I see that there has been [some discussion about it](https://github.com/tiangolo/fastapi/discussions/7292), but it doesn't look like support is coming anytime soon. Sticking with a list in JSON sounds like a better plan then.

**Comment by underdarknl** [comment]
Further work in:
https://github.com/minvws/nl-kat-coordination/issues/1125

**Review by underdarknl — APPROVED** [review]

**Pull request merged**